### DISCUSSION TITLE
How to decode event logs & topics from useWaitForTransacitons

### DISCUSSION THREAD

**Discussion by TheDojoSamurai** [wevm/wagmi#980] [Q&A] [answered]
I am wondering if anyone has code snippet to decode event logs/topics returned from `useWaitForTransactions`. I send a transaction, which results in multiple events with different topics & data. Then present that to the user in the UI. 

I am using wagmi which has `useContractEvent` which does decode the events. But on a one-by-one bases and is not transaction specific.

Using `useWaitForTransaction` hook does pass back all the events / topics, but they are encoded. 

Side proposal: Since we already provide the ABI to `useWaitForTransaction` & `useContractWrite` could return decoded events.

Any help would be much appreciated.

**Comment by RogerPodacter** [comment, marked as answer]
Something like this should work:

```javascript

const provider = useProvider()
const interface = new ethers.utils.Interface(contractABI);

const { data, isError, isLoading } = useWaitForTransaction(...)

data.logs.map(i => interface.parseLog(i))
```

**Reply by TheDojoSamurai** [reply]
Thank for this, very helpful. I did ended up using something similar. 

For my current project the logs sent back have events from multiple contracts. Some of the events are order specific, and the log index gets removed when using `interface.parseLog()`

So I am still trying to figure out a nice clean way to parse them all.

**Reply by RogerPodacter** [reply]
Good point about the multiple contracts. Does Ethers handle this when you use their contract API? I feel like it does! There must be a way.

**Reply by RogerPodacter** [reply]
Did you ever figure out a good solution here? I'm now running into the "multiple contracts" issue myself!

It's weird because when I create the interface using the ABI of an ERC721 contract it seems like that interface should be able to parse `Transfer()` events from a different ERC721 contract, but it cannot!

Thoughts?

(edit: actually having written this I believe I'm making a mistake that might be confounding the situation)

(UPDATE: if you add the correct events to the contract / ABI you're using to parse, it works)
